internal/infraestructure/github-connection/github-connection.go

```diff
@@ -2,54 +2,185 @@ package github_connection
 
 import (
 	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
 	"github.com/google/go-github/v61/github"
 )
 
-type GithubConnectionParams struct {
-	RepositoryName    string
-	Token             string
-	RepoOwner         string
-	PullRequestNumber int
+type GithubConnection struct {
+	Client            *github.Client `json:"client"`
+	RepositoryName    string         `json:"repository_name"`
+	RepoOwner         string         `json:"repo_owner,omitempty"`
+	PullRequestNumber int            `json:"pull_request_number"`
+	EventName         string         `json:"event_name"`
 }
```

Comment:
Code Review 2024-08-18 02:28:52 - New lines 7 to 12:

type GithubConnection struct {
	Client            *github.Client `json:"client"`
	RepositoryName    string         `json:"repository_name"`
	RepoOwner         string         `json:"repo_owner,omitempty"`
	PullRequestNumber int            `json:"pull_request_number"`
	EventName         string         `json:"event_name"`